### DISCUSSION TITLE
Using a singleton viewmodel in ViewMap

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno.extensions#2125] [Q&A]
Is it possible to use a singleton object for the ViewModel in the ViewMap?  In my application i want to have a single Settings ViewModel but not sure how to accomplish that with the Register.Views?

thanks

**Comment by francoistanguay** [comment]
Can you give me an idea of why you need a singleton instance? Is it because you're storing state in the ViewModel? Could you move the state in a service and register that stateful service as a singleton in DI?

**Reply by cconner100** [reply]
Thanks guys, i just moved it all to another way to do it, lets close this out.
